test/unit/AssetManagedPair.t.sol

```diff
@@ -103,47 +103,81 @@ contract AssetManagedPairTest is BaseTest
+        assertEq(_manager.getBalance(_pair, address(lToken0)), 10e18);
+        assertEq(_manager.getBalance(_pair, address(lToken1)), 10e18);
     }
 
-    function testManageReserves_KStillHolds() external
+    function testManageReserves_KStillHolds() external parameterizedTest
     {
         // arrange
         vm.prank(address(_factory));
-        _constantProductPair.setManager(_manager);
+        _pair.setManager(_manager);
 
         // liquidity prior to adjustManagement
-        _tokenA.mint(address(_constantProductPair), 50e18);
-        _tokenB.mint(address(_constantProductPair), 50e18);
-        uint256 lLiq1 = _constantProductPair.mint(address(this));
+        _tokenA.mint(address(_pair), 50e18);
+        _tokenB.mint(address(_pair), 50e18);
+        uint256 lLiq1 = _pair.mint(address(this));
 
-        _manager.adjustManagement(_constantProductPair, 50e18, 50e18);
+        _manager.adjustManagement(_pair, 50e18, 50e18);
 
         // act
-        _tokenA.mint(address(_constantProductPair), 50e18);
-        _tokenB.mint(address(_constantProductPair), 50e18);
-        uint256 lLiq2 = _constantProductPair.mint(address(this));
+        _tokenA.mint(address(_pair), 50e18);
+        _tokenB.mint(address(_pair), 50e18);
+        uint256 lLiq2 = _pair.mint(address(this));
 
         // assert
         assertEq(lLiq1, lLiq2);
     }
 
-    function testSyncManaged() external
+    function testSyncManaged_ConstantProduct() external
+    {
+        // arrange
+        vm.prank(address(_factory));
+        _constantProductPair.setManager(_manager);
+
+        address lToken0 = _constantProductPair.token0();
+        address lToken1 = _constantProductPair.token1();
+
+        _manager.adjustManagement(_constantProductPair, 20e18, 20e18);
+        _tokenA.mint(address(_constantProductPair), 10e18);
+        _tokenB.mint(address(_constantProductPair), 10e18);
+        uint256 lLiq = _constantProductPair.mint(address(this));
+
+        // sanity
+        assertEq(lLiq, 10e18); // sqrt 10e18 * 10e18
+        assertEq(_tokenA.balanceOf(address(this)), 0);
+        assertEq(_tokenB.balanceOf(address(this)), 0);
+        assertEq(_manager.getBalance(_constantProductPair, lToken0), 20e18);
+        assertEq(_manager.getBalance(_constantProductPair, lToken1), 20e18);
+
+        // act
+        _manager.adjustBalance(_constantProductPair, lToken0, 19e18); // 1e18 lost
+        _manager.adjustBalance(_constantProductPair, lToken1, 19e18); // 1e18 lost
+        _constantProductPair.transfer(address(_constantProductPair), 10e18);
+        _constantProductPair.burn(address(this));
+
+        // assert
+        assertEq(_manager.getBalance(_constantProductPair, lToken0), 19e18);
+        assertEq(_manager.getBalance(_constantProductPair, lToken1), 19e18);
+        assertLt(_tokenA.balanceOf(address(this)), 10e18);
+        assertLt(_tokenB.balanceOf(address(this)), 10e18);
+    }
+
+    function testSyncManaged_Stable() external
```

Comment:
Do these two tests still need to be separate or can be paramterized?

Reply:
in this case they need to be separate, mainly due to: 

1. the liquidity returned on L.158 and L.192 are different, because the way liquidity is calculated is diff for constant product pair vs stable pair (sqrt vs addition)
   - I have thought of using a check to see if the current `_pair` is `CP` or `SP`, but can't find ways to do it easily/reliably so decided to separate the tests for now. 
2. because the liquidity means different things for `SP` and `CP`, when we burn the same amount of LP tokens, the amount of underlying returned would be greatly different too